lib/product/types.go

```diff
@@ -121,6 +122,35 @@ func (c *Collection) validate(allowedPackageSet map[PackageName]bool) error {
 	return nil
 }
 
+type BootstrapType string
+
+var (
+	BootstrapTypeFile BootstrapType = "file"
+)
+
+type BootstrapSpec struct {
+	Type     BootstrapType `json:"type"`
+	Path     string        `json:"path"`
+	Optional bool          `json:"optional"`
+}
+
+func (b *BootstrapSpec) validate() error {
+	switch b.Type {
+	case BootstrapTypeFile:
+		if b.Path == "" {
+			return errors.Errorf("Must provide a path file type %s", BootstrapTypeFile)
+		}
+		if filepath.IsAbs(b.Path) {
+			return errors.Errorf("path must be a relative path inside the services /hab/svc/svc-name directory")
+		}
+		// TODO: validate that the path stays in the service directory
```

Comment:
I assume here you are thinking about paths like `../../../..`?

Reply:
yes